### ISSUE TITLE
Erledigungen, Fragen ...

### ISSUE THREAD

**Issue by sfrank13** [sweIhm-ws2018-19/skillproject-fr-14#29]

**Comment by sfrank13** [comment]
Allgemeine Information:
Unser Sprintziel 1 lautet ja wie folgt:

-Begrüßung: Alexa begrüßt den User
-Fächer-Gegenstandszuordnung: Gegenstände werden Fächern des Stundenplans zugeordnet
-Hilfe: Alexa assistiert dem User, falls dieser die Funktionen nicht kennt
-Einpackliste Uni: Die Einpackliste für den heutigen Tag wird bereitgestellt

Dann müssen wir auf jeden Fall noch in unserem Projects bei den "in Progress" noch die Verabschiedung raus machen und dafür dann die Einpackliste Uni rein ziehen.

Jetzt ist nur die Frage ob die Verabschiedung in Ready rein geschoben werden muss oder in den Product Backlog?

**Comment by ngaeng** [comment]
![bild1](https://user-images.githubusercontent.com/43876351/49283811-16853a80-f493-11e8-960f-a4dcd3849a2d.png)
![bild2](https://user-images.githubusercontent.com/43876351/49283812-171dd100-f493-11e8-80c3-3523248406c8.PNG)




Der statische Prototyp, da ihr den irgendwie leider nicht mehr öffnen könnt

**Comment by sfrank13** [comment]
Danke dir. 
Ich hoffe der Link geht bald wieder.

**Comment by ngaeng** [comment]
alexa öffne tasche

> Willkommen beim Skill Tasche packen!

was brauche ich

> Bei der Antwort des angeforderten Skill ist ein Problem aufgetreten

alexa stop

alexa öffne tasche

> Willkommen beim Skill Tasche packen!

was brauche ich

> Du hast heute Algorithmen und Datenstrukturen und Wahrscheinlichkeitstheorie und Statistik . Willst du heute alle Faecher besuchen ?

nein

> Welches Fach willst du nicht besuchen?

algorithmen und datenstrukturen

> Willst du noch ein Fach nicht besuchen?

nein

> Du brauchst heute Federmaeppchen

alexa stop

> Schade, dass du den Skill verlassen willst. Hoffentlich bis bald.

alexa öffne tasche

> Willkommen beim Skill Tasche packen!

was brauche ich

> Bei der Antwort des angeforderten Skill ist ein Problem aufgetreten

alexa stop

alexa öffne tasche

> Willkommen beim Skill Tasche packen!

was brauche ich

> Bei der Antwort des angeforderten Skill ist ein Problem aufgetreten



Ich hab die Eingabe und den Output aus der Developer Test-Konsole rauskopiert. Kann mir wer sagen, was ich bei der zweiten Eingabe richtig gemacht habe? Auf mich wirkt das gerade ein bisschen random, also übersehe ich wahrscheinlich was. In AWS funktioniert der Welcome Intent immer einwandfrei, auch wenn der Developer ein Problem hat.

**Comment by ngaeng** [comment]
![image](https://user-images.githubusercontent.com/43876351/49594035-45a91980-f975-11e8-8588-465380f70746.png)

Jetzt habe ich diese Meldung wenn ich im AWS debuggen will. Ein schnelles Googlen hat das Problem leider nicht gelöst.

**Comment by ngaeng** [comment]
Wahrscheinlich corrupted mein maven iwie das jar-file aber ich wirklich gar keine ahnung was sich da geändert haben könnte

**Comment by kkaiser12** [comment]
Also wegen dem ersten Problem: Ich hab da zum Teil auch so Effekte beobachtet, die mir komplett random vorkommen, vielleicht kommt da iwas im Developer nicht richtig an oder so. Eigentlich müsste der Skill ja auch bei jedem Aufruf von der Invocation neu starten, d.h. es kann nicht an einem boolean oder so liegen.

Zu dem Maven Problem: Ich hab auch keine Ahnung, an was das liegen könnte. Aber schau dir vielleicht mal, wenn du  den Maven Build machst, an, ob da irgendwelche Warnings ausgegeben werden. Hast du heute schon mal erfolgreich eine Jar reingeladen? Weil sonst könnte es evtl. daran liegen, dass ich in die POM die Library für die Mock-Tests reingeladen hab. Vielleicht musst du die noch zu deinen Libraries hinzufügen. 

UPDATE: Ich hab grade mal den neuesten Stand gepulled und bekomme den gleichen Fehler. Es scheint also am Code selbst zu liegen.

**Comment by kkaiser12** [comment]
Ok, es scheint tatsächlich an der Test-Library zu liegen. Ich hab es auskommentiert und push es nochmal, dann sollte es erstmal funktionieren. Ich schau dann nochmal, was man da anders machen muss. 

UPDATE: Ok, jetzt sollte es funktionieren, ich musste was in der POM ändern. Falls es bei dir trotzdem noch nicht geht, gib einfach Bescheid, dann schau ich es mir nochmal an.

**Comment by ngaeng** [comment]
Vielen Dank, jetzt funktioniert wieder alles 💃

**Comment by sfrank13** [comment]
![aktivitatsdiagramm_taschepacken](https://user-images.githubusercontent.com/43876401/49702003-dab94600-fbf3-11e8-89f0-7ac17c364a75.png)

**Comment by sfrank13** [comment]
Das wäre das fertige Diagramm. Ich bin mir aber echt nicht zu 100 % sicher ob dies so stimmt, aber ich habe alle möglichen Verbindungen mit eingezeichnet. Es wäre ganz gut, wenn ihr einen Blick drauf werfen könntet.

**Comment by ngaeng** [comment]
Sieht gut aus, sehr ausführlich

**Comment by kkaiser12** [comment]
Funktioniert eigentlich bei jemandem der Skill mit dem aktuellen Codestand auf Github? Ich kann keine Gegenstände hinzufügen. Es kommt immer die Meldung "Das Fach wurde nicht erkannt." Ich verstehe aber nicht wieso. Irgendwie scheint in dem AddOrRemoveItemHandler das HashSet items null zu sein.

**Comment by ngaeng** [comment]
Du musst lokal deine eigenen Accountdaten eintragen, wir ham das nur nicht mehr auf Github, weil sonst Chris' Accountdaten öffentlich sichtbar sind.

**Comment by kkaiser12** [comment]
Irgendwie habe ich das gleiche Problem auch noch mit Accountdaten, aber solange es bei euch geht, passt es ja.

**Comment by ngaeng** [comment]
Ich habs bei mir leider auch noch nicht getestet, vll weiß @Chrysotomus weiter?

**Comment by Chrysotomus** [comment]
@ngaeng @kkaiser12 nein, ich habe mich auch noch nicht aufs Testen gestürzt. Ich habe nur an der Day-Klasse umgeschrieben und diese getestet. Die funktioniert ohne Probleme. Am Alexa-Skill selber dürfte das nichts ändern. Ich bin morgen früh auf jeden Fall in der Uni, und bin wohl auch den ganzen Tag da. Wir können es uns dann gerne auch mal zusammen ansehen

**Comment by ngaeng** [comment]
hab geschaut, geht bei mir
